lib/src/main/java/at/bitfire/cert4android/UserDecisionRegistry.kt

```diff
@@ -108,10 +108,15 @@ class UserDecisionRegistry private constructor(
             customCertStore.setUntrustedByUser(cert)
 
         // continue work that's waiting for decisions
-        pendingDecisions[cert]?.iterator()?.let { iter ->
-            while (iter.hasNext()) {
-                iter.next().resume(trusted)
-                iter.remove()
+        synchronized(pendingDecisions) {
+            pendingDecisions[cert]?.iterator()?.let { iter ->
+                while (iter.hasNext()) {
+                    try {
+                        iter.next().resume(trusted)
+                    } finally {
+                        iter.remove()
+                    }
```

Comment:
Hm but try / finally without catch would still throw the Exception, wouldn't it?

Reply:
Yeah, that's true, I'll add a simple catch